doc/vim-plugin.md

```diff
@@ -197,6 +199,7 @@ let g:phpactorCustomInputListStrategy = 'InputListCustomStrategy'
 - `g:phpactorInputListStrategy`: Select a strategy for the [Input
   List](#input-list).
 - `g:phpactorCustomInputListStrategy`: Specify your own strategy.
+- `g:phpactorUseFzfForQuickfix`: Defines if fzf shoudl be used to build the quickfix, default: `v:true`.
```

Comment:
It's not really using FZF for quickfix, it's using FZF for file reference navigation.

Maybe:

```
g:phpactorFileReferenceNavigation=[quickfix|fzf|auto]`
```

Which makes it a bit more future proof.

Not related to this PR - but why do we have both `g:phpactorInputListStrategy` and `g:phpactorCustomInputListStrategy`? They could be combined into one or?

Reply:
> It's not really using FZF for quickfix, it's using FZF for file reference navigation.

That's what I explained above, for me it's not just to find references.
It's more a filter before the quickfix, so it make sense to have this name.
That's also why I said that I didn't go for the strategy option here, I can't think of another option to do what I wanted to do with fzf.
Which is allow to filter the results that we will build the quickfix with and possibly shortcut the all thing if we are interested in only one result.



> Not related to this PR - but why do we have both `g:phpactorInputListStrategy` and `g:phpactorCustomInputListStrategy`? They could be combined into one or?

The first one decide which strategy to use.
The second one allow to add new strategies.
I prefer it separate because a user might want to provide multiple new strategies and choose the one he wants on the fly depending on the functionality called.
For example with a custom mapping you could choose the strategy before calling the phpactor function:
```vim
nmap XXX let g:phpactorInputListStrategy = 'custom1' | <Plug>PhpactorAction
```

I did not check the syntax but that's the idea.

Reply:
> That's what I explained above, for me it's not just to find references.

But the quick fix is used for navigating file references, `quickfix` is just the mechanism that Phpactor uses to do that.

I think I'd prefer the strategy -- the new strategy is already quite specific - it requires `fzf.vim`, and it's not unimaginable that something interesting comes along which can be used as an alternative to `fzf`.

In other news, I notice that we don't define this config option at the top of `phpactor.vim` as we do for older options (e.g. `g:phpactorPhpBin`, `g:phpactorBranch` etc). Would it make sense to declare the options there with default values if they don't exist? I think it might be easier if all the plugin's config options are defined in one place.

Reply:
> But the quick fix is used for navigating file references, `quickfix` is just the mechanism that Phpactor uses to do that.

Yes, but I want to use it for other tasks too that's why I want this bit of code to be reusable and the name of the config variable generic enough.
Otherwise we would have the same logic with a different variable every time we would like to offer this feature to the user.

> I think I'd prefer the strategy -- the new strategy is already quite specific - it requires `fzf.vim`, and it's not unimaginable that something interesting comes along which can be used as an alternative to `fzf`.

I agree, we could do it then. It doesn't feel really YAGNI to do it so early.
But I won't fight you on that, if you want a strategy now I will do it.


For the options I don't mind, I took the habit to always use the `get()` function in case the variable is not configured.

Reply:
:+1: I think it's little effort which helps future proof us in the longer term whilst also fitting in more with the precedent of `inputListStrategy`

Then let's move the config key to the top of the phpactor.vim for now, at least for consistency.

Reply:
> Yes, but I want to use it for other tasks too

I think we may have misunderstood - when I say "navigating file references" I am referring to what `quickfix` does without Phpactor, i.e. navigating between `file/line/col` positions.

So then I think I'd agree and it would make sense for the `FileReferenceNavigation` (or similar) method could be used for any Phpactor action that invokes a quick list.